catalog/core/catalog-core-standardframework/src/test/java/ddf/catalog/impl/operations/ResourceOperationsOptionsTest.java

```diff
@@ -0,0 +1,379 @@
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsEmptyResults() throws Exception {
+        ArrayList<Result> mockResultList = new ArrayList<>();
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenReturn(
+                queryResponseMock);
+        when(queryResponseMock.getResults()).thenReturn(mockResultList);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesUnsupportedQueryException()
+            throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                UnsupportedQueryException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesFederationException() throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                FederationException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    @Test(expected = ResourceNotFoundException.class)
+    public void testGetEnterpriseResourceOptionsCatchesIllegalArgumentException() throws Exception {
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenThrow(
+                IllegalArgumentException.class);
+
+        resourceOperations.setQueryOperations(queryOperationsMock);
+        resourceOperations.getEnterpriseResourceOptions(metacard.getId(), fanoutEnabled);
+
+        verify(queryOperationsMock).query(any(), any(), anyBoolean(), anyBoolean());
+    }
+
+    // getResourceOptions
+    @Test
+    public void testGetResourceOptions()
+            throws ResourceNotFoundException, FederationException, UnsupportedQueryException {
+
+        Result resultMock = mock(Result.class);
+
+        ArrayList<Result> mockResultList = new ArrayList<>();
+        mockResultList.add(resultMock);
+
+        when(queryOperationsMock.query(any(), any(), anyBoolean(), anyBoolean())).thenReturn(
```

Comment:
The code from 183-191 looks like a structure that is repeated several times. Is it possible to put this into a helper method instead of repeating the structure?